powerapi-cli/src/main/scala/org/powerapi/app/Application.scala

```diff
@@ -0,0 +1,222 @@
+
+class FileDisplay(filepath: String) extends org.powerapi.reporter.FileDisplay {
+  override lazy val output = Resource.fromFile(filepath)
+}
+
+/**
+ * PowerAPI CLI.
+ *
+ * @author <a href="mailto:maxime.colmant@gmail.com">Maxime Colmant</a>
+ * @author <a href="mailto:l.huertas.pro@gmail.com">Loïc Huertas</a>
+ */
+object PowerAPI extends App {
+  val modulesR = """(cpu-simple|cpu-dvfs|libpfm-core|libpfm-core-process|powerspy)(,(cpu-simple|cpu-dvfs|libpfm-core|libpfm-core-process|powerspy))*""".r
+  val aggR = """max|min|geomean|logsum|mean|median|stdev|sum|variance""".r
+  val durationR = """\d+""".r
+  val pidR = """(\d+)""".r
+  val appR = """(.+)""".r
+
+  @volatile var powerMeters = Seq[PowerMeter]()
+  @volatile var monitors = Seq[PowerMonitoring]()
+
+  val shutdownHookThread = scala.sys.ShutdownHookThread {
+    println("It's the time for sleeping! ...")
+
+    monitors.foreach(monitor => monitor.cancel())
+    monitors = Seq()
+    powerMeters.foreach(powerMeter => powerMeter.shutdown())
+    powerMeters = Seq()
+  }
+
+  def validateModules(str: String) = str match {
+    case modulesR(_*) => true
+    case _ => false
+  }
+
+  implicit def modulesStrToPowerModules(str: String): Seq[PowerModule] = {
+    (for(module <- str.split(",")) yield {
+      module match {
+        case "cpu-simple" => CpuSimpleModule()
+        case "cpu-dvfs" => CpuDvfsModule()
+        case "libpfm-core" => LibpfmCoreModule()
+        case "libpfm-core-process" => LibpfmCoreProcessModule()
+        case "powerspy" => PowerSpyModule()
+      }
+    }).toSeq
+  }
+
+  def validateAgg(str: String): Boolean = str match {
+    case aggR(_*) => true
+    case _ => false
+  }
+
+  implicit def aggStrToAggFunction(str: String): Seq[Power] => Power = {
+    str match {
+      case "max" => MAX
+      case "min" => MIN
+      case "geomean" => GEOMEAN
+      case "logsum" => LOGSUM
+      case "mean" => MEAN
+      case "median" => MEDIAN
+      case "stdev" => STDEV
+      case "sum" => SUM
+      case "variance" => VARIANCE
+    }
+  }
+
+  def validateDuration(str: String): Boolean = str match {
+    case durationR(_*) => true
+    case _ => false
+  }
+
+  implicit def targetsStrToTargets(str: String): Seq[Target] = {
+    val strTargets = if(str.split(",").contains("all")) {
+      "all"
+    }
+    else str
+
+    (for(target <- strTargets.split(",")) yield {
+      target match {
+        case "" => Process(ManagementFactory.getRuntimeMXBean.getName.split("@")(0).toInt)
+        case "all" => All
+        case pidR(pid) => Process(pid.toInt)
+        case appR(app) => Application(app)
+      }
+    }).toSeq
+  }
+
+  def printHelp(): Unit = {
+    val str =
+      """
+        |PowerAPI (c) Spirals Team"
+        |
+        |Build a software-defined power meter. Don't forget to configure correctly the modules (see the documentation).
+        |
+        |usage: ./powerapi-cli modules [cpu-simple|cpu-dvfs|libpfm-core|libpfm-core-proces|powerspy, ...]
```

Comment:
I saw that like an entry (as duration), thus we can use the option with -- after. It is possible to plug different modules also on the same PowerMeter.